modules/apps/commerce/commerce-channel-web/src/main/java/com/liferay/commerce/channel/web/internal/display/context/CommerceChannelCountryDisplayContext.java

```diff
@@ -0,0 +1,163 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.commerce.channel.web.internal.display.context;
+
+import com.liferay.commerce.channel.web.internal.item.selector.criterion.CountryItemSelectorCriterion;
+import com.liferay.commerce.product.display.context.helper.CPRequestHelper;
+import com.liferay.commerce.product.model.CommerceChannel;
+import com.liferay.commerce.product.model.CommerceChannelRelModel;
+import com.liferay.commerce.product.service.CommerceChannelRelService;
+import com.liferay.commerce.product.service.CommerceChannelService;
+import com.liferay.frontend.taglib.clay.servlet.taglib.util.CreationMenu;
+import com.liferay.item.selector.ItemSelector;
+import com.liferay.item.selector.ItemSelectorCriterion;
+import com.liferay.item.selector.criteria.UUIDItemSelectorReturnType;
+import com.liferay.petra.function.transform.TransformUtil;
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.dao.orm.QueryUtil;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.language.LanguageUtil;
+import com.liferay.portal.kernel.portlet.LiferayPortletResponse;
+import com.liferay.portal.kernel.portlet.RequestBackedPortletURLFactory;
+import com.liferay.portal.kernel.portlet.RequestBackedPortletURLFactoryUtil;
+import com.liferay.portal.kernel.portlet.url.builder.PortletURLBuilder;
+import com.liferay.portal.kernel.security.permission.ActionKeys;
+import com.liferay.portal.kernel.security.permission.resource.ModelResourcePermission;
+import com.liferay.portal.kernel.util.HashMapBuilder;
+import com.liferay.portal.kernel.util.ParamUtil;
+import com.liferay.portal.kernel.util.StringUtil;
+
+import java.util.HashMap;
+
+import javax.servlet.http.HttpServletRequest;
+
+/**
+ * @author Stefano Motta
+ */
+public class CommerceChannelCountryDisplayContext {
+
+	public CommerceChannelCountryDisplayContext(
+		ModelResourcePermission<CommerceChannel>
```

Comment:
Because the modelPermission should go always  as last parameter? Because by name  (commerceChannelModelResourcePermission) the order should be correct

Reply:
I'm not sure @alessiorendina what do you think?

Reply:
This should be alphabetically sorted, so it seems ok to me